webapp/channels/src/components/advanced_text_editor/send_button/send_post_options/index.tsx

```diff
@@ -0,0 +1,175 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import classNames from 'classnames';
+import React, {useCallback, useMemo} from 'react';
+import {FormattedMessage, useIntl} from 'react-intl';
+
+import ChevronDownIcon from '@mattermost/compass-icons/components/chevron-down';
+import type {SchedulingInfo} from '@mattermost/types/schedule_post';
+
+import * as Menu from 'components/menu';
+import Timestamp from 'components/timestamp';
+
+import './style.scss';
+
+type Props = {
+    disabled?: boolean;
+    onSelect: (e: React.FormEvent, schedulingInfo: SchedulingInfo) => void;
+}
+
+export function SendPostOptions({disabled, onSelect}: Props) {
+    const {formatMessage} = useIntl();
+
+    const handleOnSelect = useCallback((e: React.FormEvent, scheduledAt: number) => {
+        const schedulingInfo: SchedulingInfo = {
+            scheduled_at: scheduledAt,
+        };
+
+        onSelect(e, schedulingInfo);
+    }, [onSelect]);
+
+    const coreMenuOptions = useMemo(() => {
+        const today = new Date();
+        today.setHours(9, 0, 0, 0); // 9 AM
+        const unixTimestamp = today.getTime();
+
+        const timeComponent = (
+            <Timestamp
+                value={unixTimestamp}
+                useDate={false}
+            />
+        );
+
+        const tomorrow9amTime = new Date();
+        tomorrow9amTime.setHours(9, 0, 0, 0);
+        tomorrow9amTime.setDate(today.getDate() + 1);
+
+        const optionTomorrow = (
+            <Menu.Item
+                onClick={(e) => handleOnSelect(e, tomorrow9amTime.getTime())}
+                labels={
+                    <FormattedMessage
+                        id='create_post_button.option.schedule_message.options.tomorrow'
+                        defaultMessage='Tomorrow at {9amTime}'
+                        values={{'9amTime': timeComponent}}
```

Comment:
Are you sure we need to use the timestamp here? I would've expected that this would just be part of the translation string since the translator would know how to translate it. It seems weird to use the `Timestamp` component here when it includes so much logic in it which isn't otherwise needed here

Reply:
We need timestamp here as we need to show the time as per user's configured timezone and based on the user's preference of 12 or 24 hour time format.

Reply:
Shouldn't it already be 9am in their time zone because it comes from a `Date` object? That's a good point about the 12 or 24 hour time format though.

It does seem a bit funny that we're passing the same 9am tomorrow `Timestamp` component into the `FormattedMessage` for "Monday at 9:00", but that shouldn't cause any problems